### PR TITLE
Setup map

### PR DESCRIPTION
**Pull request by okg00dbye**
image
you should see this at http://localhost:8000/map/demo . it's also draggable

- the unit block is 1x1x1
- TODO: replace hardcoded color value, not sure why MuiTheme doesn't work
- let me know if anyone faces `Expression produces a union type that is too complex to represent.` It's an ulu error that doesn't seem to have a good fix :') (https://stackoverflow.com/questions/68692230/ts-expression-produces-a-union-type-that-is-too-complex-to-represent-with-materi) I patched @react-three/drei, hope it works correctly

The compile times are still unbearable ;-; yarn add can take almost 7 minutes and yarn start can take up to 10 minutes...hot reload is still 20+ seconds haishhh I'm retreating back to code sandbox bye bye

### CODE REVIEW

**Comment by sebastiantoh** [comment]
> TODO: replace hardcoded color value, not sure why MuiTheme doesn't work

This might be because the import was wrong. I've pushed a commit to fix it

File: frontend/src/components/map/Character.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useLoader } from '@react-three/fiber';
+import { OBJLoader } from 'three/examples/jsm/loaders/OBJLoader';
+import { MTLLoader } from 'three/examples/jsm/loaders/MTLLoader';
+
+import { MapComponent } from './MapComponent';
+
+const Character: React.FC = () => {
+  const materials = useLoader(MTLLoader, '/models/astronaut.mtl');
```
Comment (sebastiantoh) [inline frontend/src/components/map/Character.tsx:9]: the code for this seems identical to that of `Model`. Is it possible to modify `Character` to use `Model`?

**Review by sebastiantoh — COMMENTED** [review]
looks good!

File: frontend/src/components/map/Character.tsx (hunk 1 above)
Comment (jialin7878) [reply frontend/src/components/map/Character.tsx:9]: Ah me separated because it's likely they will diverge in the future, e.g special character animations

**Pull request merged**